buildscripts/CommonAssemblyInfo.cs

```diff
@@ -12,12 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System;
+using System.Runtime.InteropServices;
 using System.Reflection;
+using System.Security;
+
+[assembly: AssemblyProduct(
+	#if DOTNET45 // order is important
+	"Castle Core for .NET Framework 4.5"
+	#elif DOTNET40
+	"Castle Core for .NET Framework 4.0"
+	#elif DOTNET35
+	"Castle Core for .NET Framework 3.5"
+	#else
+	"Castle Core for .NET Core 1.0"
+	#endif
+)]

```

Comment:
All DLLs will have same name, including the logging plugins.

(Also no "compact framework" for 4.0, which doesn't worry me a bit.)